src/normalize/normalizer.rs

```diff
@@ -104,14 +106,45 @@ impl BuildIdReader for DefaultBuildIdReader {
     fn read_build_id_from_elf(path: &Path) -> Result<Option<Vec<u8>>> {
         let file = File::open(path)?;
         let parser = ElfParser::open_file(file)?;
-        Self::read_build_id(&parser)
+        if let Ok(Some(build_id)) = Self::read_build_id_from_section_name(&parser) {
+            Ok(Some(build_id))
+        } else if let Ok(Some(build_id)) = Self::read_build_id_from_notes(&parser) {
+            Ok(Some(build_id))
+        } else {
+            Ok(None)
+        }
+    }
+
+    // Iterate over all note sections to find one of type NT_GNU_BUILD_ID
+    fn read_build_id_from_notes(parser: &ElfParser) -> Result<Option<Vec<u8>>> {
+        let shdrs = parser.section_headers()?;
+        for shdr in shdrs.iter() {
+            if shdr.sh_type == elf::types::SHT_NOTE {
+                let offset = shdr.sh_offset;
+                let size = shdr.sh_size;
+                let mut bytes = parser
+                    .section_data_from_offset(offset as usize, size as usize)
```

Comment:
Can we iterate over `shdrs.iter().enumerate()` and then use the existing index based section-data lookup? It's not clear to me why we would want to introduce yet another way?